octodns/provider/ns1.py

```diff
@@ -35,11 +37,38 @@ def __init__(self, id, api_key, *args, **kwargs):
         self._client = NSONE(apiKey=api_key)
 
     def _data_for_A(self, _type, record):
-        return {
+        # record meta (which would include geo information is only
+        # returned when getting a record's detail, not from zone detail
+        geo = {}
+        data = {
             'ttl': record['ttl'],
             'type': _type,
-            'values': record['short_answers'],
         }
+        values, codes = [], []
+        if 'answers' not in record:
+            values = record['short_answers']
+        for answer in record.get('answers', []):
+            meta = answer.get('meta', {})
+            if meta:
+                country = meta.get('country', [])
+                us_state = meta.get('us_state', [])
+                ca_province = meta.get('ca_province', [])
```

Comment:
Is it possible for multiple of these to be non-default? E.g.

```json
{
  "country": "US",
  "us_state": "WA"
}
```

If so it seems like this would set both `geo['US']` and `geo['US-WA']`. If only `us_state` would be set and not  country for cases where a US state was targeted then this would work, in which case a comment to that effect would probably make sense.

Reply:
It's possible, but we collapse things, so if we receive both country of US and state of WA, we only consider the more specific state of WA.  will add comment clarifying that.